common/filetoolkitwithundo.cpp

```diff
@@ -0,0 +1,204 @@
+#include "filetoolkitwithundo.h"
+
+#include "directorycleanuphandler.h"
+#include "filecleanuphandler.h"
+#include "mountcleanuphandler.h"
+
+namespace softwarecontainer {
+
+FileToolkitWithUndo::~FileToolkitWithUndo()
+{
+    bool success = true;
+    // Clean up all created directories, files, and mount points
+    for (auto it = m_cleanupHandlers.rbegin(); it != m_cleanupHandlers.rend(); ++it) {
```

Comment:
Is it possible to use for-each style loop here? In that case it would look nicer imho. But then we would have to clear() the vector afterwards.